### DISCUSSION TITLE
Terminal UI for watch scripts, similar to Turborepo 2.0

### DISCUSSION THREAD

**Discussion by romannurik** [nrwl/nx#29098] [Feature Requests]
Is anyone looking into building something like Turborepo's [Terminal UI](https://turbo.build/blog/turbo-2-0#new-terminal-ui) for NX, which is super super useful for watch scripts?

This is what it looks like:
![turbo](https://github.com/user-attachments/assets/fc464bcb-4c5a-4ead-a952-7b8430cb0de9)

I've tried making my own, powered by `blessed.js`, and delivered as a custom executor, but because of #8167 (no custom logging), the only real way to do it is spawn multiple `nx` subprocesses, which is super flaky (likely due to cache read/write collisions, etc)

**Comment by romannurik** [comment]
I threw my attempt at this over onto npm, here:
https://www.npmjs.com/package/nx-devui

I do wish this were built-in though, or at least less hacky.

**Reply by sceptre12** [reply]
Ohh wow did know they had that feature! That looks amazing

**Comment by puku0x** [comment]
> Nx Terminal UI (TUI) coming in Nx v21

https://x.com/MrJamesHenry/status/1907172290493628662

https://github.com/nrwl/nx/pull/30565
